### PR TITLE
Battle turns

### PR DESCRIPTION
**Pull request by EnzoCoronel**

### CODE REVIEW

**Review by vickyad — APPROVED** [review]

File: src/index.js [hunk 1]
```diff
@@ -102,59 +109,272 @@ const menu = async () => {
...
+      while (player.hp > 0 && enemy.hp > 0) {
+        console.log(`${player.name}: ${player.hp}\n${enemy.name}: ${enemy.hp}`);
+        if (turn) {
+          if (playerDmg >= 0) {
+            enemy.hp -= playerDmg;
+            console.log(`You dealt ${playerDmg} dmg to the bug\n`);
+          } else {
+            console.log("Your dmg is null against this bug!\n");
+          }
+          turn--;
```
Comment (caiorrs) [inline src/index.js:132]: considering turns can only be 0 and 1, why not guarantee that with `turn = 0` and `turn = 1`?

File: src/index.js [hunk 2]
```diff
@@ -102,59 +109,272 @@ const menu = async () => {
...
+  };
+
+  const mission = async (gamer, task, enemies) => {
+    clear();
+    let pDmg,
+      bDmg,
+      enemyList = [1],
+      weak,
+      b4battle = gamer.hp;
+    task.bugs.forEach((enemyId) => {
```
Comment (caiorrs) [inline src/index.js:164]: I'm almost sure you could achieve the same result using `map`

File: src/index.js [hunk 3]
```diff
@@ -102,59 +109,272 @@ const menu = async () => {
...
+      let enemyInfo = enemies.find((bug) => bug.id == enemyId);
+      enemyList.push(enemyInfo);
+    });
+    enemyList.shift();
+    enemyList.forEach(async (foe) => {
```
Comment (caiorrs) [inline src/index.js:169]: I would like to ask the opinion of other devs about the usage of `forEach` with an async callback, it does not  seem a good idea

File: src/index.js [hunk 4]
```diff
@@ -102,59 +109,272 @@ const menu = async () => {
...
+        );
+      }
+    });
+    if (gamer.hp > 0 && weak != 1) {
+      console.log(`You won ${task.reward} gold!\n`);
+      gamer.gold += task.reward;
+      await patchApi({ gold: gamer.gold }, `characters/${gamer.id}`);
+    } else if (gamer.hp <= 0) {
+      if (gamer.equipments.length > 0) {
+        let remove = Math.floor(Math.random() * gamer.equipments.length);
```
Comment (caiorrs) [inline src/index.js:188]: Suggestion, create a function to generate random numbers, might be a little easier to use and understand

File: src/index.js [hunk 5]
```diff
@@ -102,59 +109,272 @@ const menu = async () => {
...
+        } else {
+          mission(playerInfo, questList[select - 1], bugList);
+        }
+      }
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  const store = async (costumer) => {
```
Comment (caiorrs) [inline src/index.js:253]: typo? Did you mean `customer`?

**Pull request merged**